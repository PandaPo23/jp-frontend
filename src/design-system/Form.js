import React from 'react';

import Grid from './Grid';
import system from '../utils/System';

const StyledForm = system('StyledForm', {
  extend: Grid,
  as: 'form',
  gridGap: '0.5rem',
  gridTemplateColumns: 'auto 1fr',
});

const Form = ({ children, ...props }) => (
  <StyledForm {...props}>{children}</StyledForm>
);

export default Form;
