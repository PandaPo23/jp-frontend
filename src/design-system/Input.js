import React from 'react';

import system from '../utils/System';
import Box from './Box';

// import { themeGet } from 'styled-system';

const StyledInput = system(
  'Input',
  {
    extend: Box,
    as: 'input',
    type: 'text',
    boxShadow: 0,
    borderRadius: 'sm',
    bg: 'surface',
    flex: 1,
    border: 1,
    p: 2,
    width: 1,
    value: '',
  },
  (props) => ({
    appearance: 'none',
    // ':focus': {
    //   border: props.border && themeGet(`borders.${props.border}`)(props),
    // },
    // '::placeholder': {
    //   color: themeGet(`colors.misc.muted`)(props),
    // },
    // '::ms-clear': {
    //   display: 'none',
    // },
  })
);

export const Input = React.forwardRef(
  ({ autoFocus = false, onChange, ...props }, ref) => (
    <StyledInput
      ref={ref}
      autoFocus={autoFocus}
      onChange={onChange ? (event) => onChange(event.target.value) : undefined}
      {...props}
    />
  )
);

export default Input;
