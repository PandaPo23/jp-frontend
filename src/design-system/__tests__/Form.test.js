import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../Input';
import Form from '../Form';

describe('Form component', () => {
  it('should render form component', () => {
    const wrapper = mount(<Form />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render component with children', () => {
    const onChange = jest.fn();
    const component = mount(
      <Form>
        <Input onChange={onChange} />
      </Form>
    );
    expect(component.find(Input).exists()).toBe(true);
  });
  it('should simulate submit', () => {
    const fakeSubmit = jest.fn();
    const onChange = jest.fn();
    const component = mount(
      <Form onSubmit={fakeSubmit}>
        <Input onChange={onChange} />
      </Form>
    );
    component.find('form').simulate('submit');
    expect(fakeSubmit).toBeCalled();
  });
});
