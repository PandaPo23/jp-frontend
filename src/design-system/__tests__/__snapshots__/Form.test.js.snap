// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render form component 1`] = `
.emotion-0 {
  display: grid;
  grid-gap: 0.5rem;
  grid-template-columns: auto 1fr;
}

<Form>
  <StyledForm
    as="form"
    display="grid"
    gridGap="0.5rem"
    gridTemplateColumns="auto 1fr"
  >
    <form
      className="emotion-0"
    />
  </StyledForm>
</Form>
`;
